### ISSUE TITLE
Stuck on loading….

### ISSUE THREAD

**Issue by plasterer41** [Blackymas/NSPanel_HA_Blueprint#485]
![image](https://user-images.githubusercontent.com/47918336/226439971-03456cf6-38fb-4c98-a68f-213101236369.jpg)
![image](https://user-images.githubusercontent.com/47918336/226440160-3492349e-e8c1-4a49-a2be-2c9b88d148cf.jpg)

**Comment by damjangr** [comment]
I have the same problem

**Comment by henrykuijpers** [comment]
Same here, I'm seeing the same screen, saying:
"TFT: 3.2"
"ESPHome: 3.2"
"Blueprint: " (nothing)

"Please wait... The process can take several seconds!"

It's not making any progress, at all.

**Comment by henrykuijpers** [comment]
@Blackymas Do you have any idea what this would mean? In my case, the blueprint doesn't mention the version number, in case of @plasterer41 it does.
What would it mean? In my case, it doesn't know which blueprint to load/show, which then makes it wait forever?

I do have the device name set to "nspanel-1" (with a hyphen). I'm reading elsewhere that hyphens are not allowed in the device name.

**Comment by henrykuijpers** [comment]
Oh that is very nice @damjangr ! So it might have to do with incorrect configuration then. Would be nice if we could find logging of this somewhere, that would indicate the issue. :)

**Comment by Blackymas** [comment]
Hello,

I have written this very often here. Please use the search next time.

If you get stuck on the boot page or it jumps to home page and stops in loading something is wrong in your blueprint.

Either wrong or incomplete entities are selected or the entity has an undefined or wrong status.

The best is to delete the blueprint and create a new one and fill in only the first points that are not optional.

If you have already used 3.1 check all fields. With version 3.2 also existing fields have changed which could be a problem now.

In addition, you can use the point trace in the upper right corner of your automation to see where it gets stuck.

See here:

https://github.com/Blackymas/NSPanel_HA_Blueprint/wiki/(EN)-HowTo---All-important-thing-you-should-know#1-how-do-I-create-issues-when-I-have-a-problem

**Comment by henrykuijpers** [comment]
@Blackymas  It seems to be stuck at "Executed: 20 March 2023 at 21:25:00
Result:
result: false
entities:
  - switch.nspanel-1_nextion_inited"

Do you have an idea of how to fix it? I think it has to do with the display? Also, the blueprint version still isn't showing, even though I did fill in the various required fields.

**Comment by henrykuijpers** [comment]
Just did a reflash with `nspanel_1` as the device name, instead of `nspanel-1`; Now it actually got a little bit further.

```
aioesphomeapi.core.TimeoutAPIError: Timeout waiting for response for <class 'api_pb2.PingRequest'> after 10.0s
INFO Disconnected from ESPHome API for nspanel_1.local
WARNING Disconnected from API
INFO Successfully connected to nspanel_1.local
[21:43:43][D][nspanel:961]: no page-jump
[21:43:44][D][number:054]: 'nspanel_1 Display Brightness' - Setting number value
[21:43:44][D][number:113]:   New number value: 100.000000
[21:43:44][D][number:012]: 'nspanel_1 Display Brightness': Sending state 100.000000
[21:43:44][D][number:054]: 'nspanel_1 Display Brightness Dimdown' - Setting number value
[21:43:44][D][number:113]:   New number value: 10.000000
[21:43:44][D][number:012]: 'nspanel_1 Display Brightness Dimdown': Sending state 10.000000
[21:43:45][D][switch:056]: 'nspanel_1 Nextion inited': Sending state ON
[21:44:15][D][sensor:127]: 'ntc_source': Sending state 1.66300 V with 2 decimals of accuracy
[21:44:15][D][resistance:039]: 'resistance_sensor' - Resistance 11377.9Ω
[21:44:15][D][sensor:127]: 'resistance_sensor': Sending state 11377.88672 Ω with 1 decimals of accuracy
[21:44:15][D][ntc:026]: 'nspanel_1 Temperature' - Temperature: 22.1°C
[21:44:15][D][sensor:127]: 'nspanel_1 Temperature': Sending state 22.12298 °C with 1 decimals of accuracy
[21:44:23][D][sensor:127]: 'nspanel_1 uptime': Sending state 51.43900 s with 0 decimals of accuracy
[21:44:29][D][esp32.preferences:113]: Saving 2 preferences to flash...
[21:44:29][D][esp32.preferences:142]: Saving 2 preferences to flash: 2 cached, 0 written, 0 failed
[21:44:33][D][sensor:127]: 'nspanel_1 RSSI': Sending state -66.00000 dBm with 0 decimals of accuracy
[21:45:15][D][sensor:127]: 'ntc_source': Sending state 1.66400 V with 2 decimals of accuracy
[21:45:15][D][resistance:039]: 'resistance_sensor' - Resistance 11391.7Ω
[21:45:15][D][sensor:127]: 'resistance_sensor': Sending state 11391.68750 Ω with 1 decimals of accuracy
[21:45:15][D][ntc:026]: 'nspanel_1 Temperature' - Temperature: 22.1°C
[21:45:15][D][sensor:127]: 'nspanel_1 Temperature': Sending state 22.09622 °C with 1 decimals of accuracy
[21:45:23][D][sensor:127]: 'nspanel_1 uptime': Sending state 111.43900 s with 0 decimals of accuracy
[21:45:33][D][sensor:127]: 'nspanel_1 RSSI': Sending state -65.00000 dBm with 0 decimals of accuracy
[21:46:15][D][sensor:127]: 'ntc_source': Sending state 1.66200 V with 2 decimals of accuracy
[21:46:15][D][resistance:039]: 'resistance_sensor' - Resistance 11364.1Ω
[21:46:15][D][sensor:127]: 'resistance_sensor': Sending state 11364.10254 Ω with 1 decimals of accuracy
[21:46:15][D][ntc:026]: 'nspanel_1 Temperature' - Temperature: 22.1°C
[21:46:15][D][sensor:127]: 'nspanel_1 Temperature': Sending state 22.14974 °C with 1 decimals of accuracy
[21:46:23][D][sensor:127]: 'nspanel_1 uptime': Sending state 171.43900 s with 0 decimals of accuracy
[21:46:31][I][ota:113]: Boot seems successful, resetting boot loop counter.
[21:46:31][D][esp32.preferences:113]: Saving 1 preferences to flash...
[21:46:31][D][esp32.preferences:142]: Saving 1 preferences to flash: 0 cached, 1 written, 0 failed
[21:46:33][D][sensor:127]: 'nspanel_1 RSSI': Sending state -63.00000 dBm with 0 decimals of accuracy
```

It seems to be doing a successful boot, however, it still doesn't show a Blueprint version. It seems to say "no page-jump"? But I don't know if that would be an issue?

It does seem to correctly catch up with temperature etc events and send them. If I press the physical buttons on the device, the actions do appear in HA. So that is good.

**Comment by Blackymas** [comment]
If you have flashed the NSpanel with the new name you have to do all the other steps again.

Add device under HA and create new blueprint.

It is best to delete the HA device and the Blueprint before.

have a look here:
https://github.com/Blackymas/NSPanel_HA_Blueprint/wiki/(EN)-First-Steps---Installation-and-Setup

**Comment by 86Justdave** [comment]
i have deleted and set up the blueprint again, still no luck, this is my trace log?

only basic entities filled in.

alias: NSPanel Configuration
description: bedroom panel
use_blueprint:
  path: Blackymas/nspanel_blueprint.yaml
  input:
    language: ENG
    nspanel_name: bedroompanel
    weather_entity: weather.home

and still showing esphome.bedroompanel_set_settings_entity as an unknown service?

![trace](https://user-images.githubusercontent.com/123274014/226462540-7bb0b6f3-5eb1-4964-9ed0-01116b1851dd.jpg)

**Comment by Blackymas** [comment]
something is still wrong with your name.

What I would recommend:

1. delete the blueprint
2. delete your NSPanel in HA under Esphome
3. delete the NSpanel config file in ESPhome

4. create a new esphome file with a correct name without special characters or "-".
5. flash your Nspanel again with dam cable (not wireless)
6. add the device in HA again
7. create a new blueprint

The error you have indicates that something is wrong with the name and therefore it can not find the service.

The service can be found by the following command.

 command_set_settings_entity: "esphome.{{ nspanel_name }}_set_settings_entity"

nspanel_name -> is the name you enter in bluebrint and it doesn't seem to match the name of the esp32 service.

**Comment by 86Justdave** [comment]
@Blackymas  deleted everything and reinstalled 
its now all working as should.
thank you very much

**Comment by henrykuijpers** [comment]
Here, it's also working now!

**Comment by Blackymas** [comment]
@henrykuijpers what was your issue?

**Comment by Blackymas** [comment]
@86Justdave thanks for feedback ;)

**Comment by SunilKhusal** [comment]
I have a similar issue, stuck on loading...

The initial screen shows:
"TFT: 3.2"
"ESPHome: 3.1.0"
"Blueprint: 3.2"

How do I update the ESPHome to 3.2?

**Comment by henrykuijpers** [comment]
@SunilKhusal you should go and check the traces, it should be stuck somewhere, maybe on one of your entities. It actually did find your blueprint, which is good. 
You can also try debug logging on the esphome device in HA. Maybe it will show you some details.

**Comment by Blackymas** [comment]
@SunilKhusal 
you find all in our wiki
https://github.com/Blackymas/NSPanel_HA_Blueprint/wiki/(EN)-HowTo---All-important-thing-you-should-know

**Comment by SunilKhusal** [comment]
@Blackymas Thank you, it was one of my sensors that had gone offline. Removed each sensor at a time and found the culprit.  Surely the blueprint should still work should any of the sensors become available for whatever reason?

**Comment by henrykuijpers** [comment]
I agree, the panel should show the entity as unavailable then, I would say. 

Was that really the issue though?

**Comment by Blackymas** [comment]
yes we check for anavailable. i need to know which entity exactly caused the error or which field in the blueprint

**Comment by albkk** [comment]
I have followed all the advice above, but it looks like my problem is slightly different. After flashing the latest version of esphome, my nspanels don't connect to my wifi (the AP is right next to the nspanel). In the boot screen the IP address is not showing and I can't see the devices in my router. I therefore assume I cannot be a problem with the blueprint (which I have created new anyway as suggested above, as well as all the other setup steps started from scratch)

Before the upgrade, the nspanels were all working perfectly with the previous verson of ESP and 3.1 blueprint.

Has anybody else come across the same problem?

**Comment by henrykuijpers** [comment]
This does sound like an issue outside of the HA/blueprint setup though. Are you sure your wifi is correctly configured? Ssid & password?

**Comment by albkk** [comment]
Indeed, my case is probably not realted to HA/blueprint setup. I have still a few panels running on ESPHome 3.1 and I can see how they first get the IP and then the blueprint loads up. I don't get to that stage in the panels flashed with ESPHome3.2

No changes were made to the configuration. SSIDs are kept in secrets are have not changed. I have also tried with a different ssid just to make sure it is not the router blocking them, but still does not work. It must have something to do with the ESPHome 3.2 upgrade. Is there any way to downgrade to 3.1 via manual flashing?

**Comment by jjnj023** [comment]
I have an issue regarding the loading too. However for me it happens when I turn on a switch. When turning it of it doesn't happen, when turning on it does. [Youtube link with a video showing the issue](https://youtube.com/shorts/RkTtdqUxAOY?feature=share)

**Comment by deejaybeam** [comment]
looks like your "computer" button became an unsupported state when turning off, like "unknown" or unavailable.
You could open the developer-tools and have a look at the state of your switch, or look at the traces of the automation.
Here you can see the trigger and the old an new states which triggers the automation

**Comment by scutler81** [comment]
I'm getting loading on the home page as well. I've tried deleting the blueprint, device, and everything from ESPHome. Created a new entry in ESPHome and flashed the device using the wired connection. Powered it up and finished adding it to HomeAsistant and then created a new blueprint. It still goes to the home page and just says loading.

The only fields I've filled in on the blueprint are:

- ESPhome Node Name
- Language for NSPanel
- Weather Integration (AccuWeather)
- Weather entity from HA (my AccuWeather entity)
I've tried the default weather integration with some of the other weather providers as well but nothing changes.

I also tried setting up the Left and right button to just turn on the built in relays. When I press the left/right buttons I can hear the relay clicking so at least that works.

All of the traces seem to run all the way through but it still showing loading.

I've done the same thing on 3 other NSPanels and they all have the same problem

Any other suggestions?

**Comment by Blackymas** [comment]
@scutler81 have you installed v3.2.1?

**Comment by scutler81** [comment]
Hi @Blackymas,

Yes, this is using v3.2.1 of the yaml in a nested configuration. And I also updated the to the latest TFT file on all devices. As far as my setup goes my YAML files just contain the substitutions section and then has nspanel_esphome.yaml as a nested configuration. 

The version of nspanel_esphome.yaml that I have has some minor customizations. I added domain and use_address in the wifi section. And then I am using the advanced configuration to specify different passwords for the API, web_server, and ota.

Could that be part of the problem?

[nspanel_esphome_yaml diff report.zip](https://github.com/Blackymas/NSPanel_HA_Blueprint/files/11082940/nspanel_esphome_yaml.diff.report.zip)

**Comment by deejaybeam** [comment]
> Hi @Blackymas,
> 
> Yes, this is using v3.2.1 of the yaml in a nested configuration. And I also updated the to the latest TFT file on all devices. As far as my setup goes my YAML files just contain the substitutions section and then has nspanel_esphome.yaml as a nested configuration.
> 
> The version of nspanel_esphome.yaml that I have has some minor customizations. I added domain and use_address in the wifi section. And then I am using the advanced configuration to specify different passwords for the API, web_server, and ota.
> 
> Could that be part of the problem?
> 
> [nspanel_esphome_yaml diff report.zip](https://github.com/Blackymas/NSPanel_HA_Blueprint/files/11082940/nspanel_esphome_yaml.diff.report.zip)

Hi
I use nested config too, this is also prepared as you can see.
In my config I also use api/web/ota - passwords, also prepared

But what I don´t understand is, why do you use "use_address" instead of a fixed/static ip-address?
use_address, is a "helper", If you need to change a static IP via OTA to another one, or rename device: https://esphome.io/components/esphome.html#esphome-changing-node-name

**Comment by scutler81** [comment]
@deejaybeam,  At some point all of my devices started showing offline in esphome even though they were online and I could control them, view logs, and wirelessly install updates. Not sure what caused it. It showed up after an update to esphome. After some Googling I found some other people reporting similar issues and they fixed it either by using static addresses or with use_address. 

@edwardtfn I powered up one of the panels and saw these two messages:

Logger: aioesphomeapi.connection
Source: runner.py:179
First occurred: March 24, 2023 at 4:13:03 PM (150 occurrences)
Last logged: 12:47:32 PM

espnspoffice @ 172.22.66.122: Connection error occurred: [Errno 104] Connection reset by peer
espnspoffice @ 172.22.66.122: Connection error occurred: Ping response not received after 90.0 seconds
espnspoffice @ 172.22.66.122: Connection error occurred: Timeout waiting for response for <class 'api_pb2.ConnectRequest'> after 30.0s
esp-switches-frontroom @ 172.22.66.111: Connection error occurred: [Errno 104] Connection reset by peer
espnspoffice @ 172.22.66.122: Connection error occurred: EOF received


Logger: homeassistant.components.automation.espnspoffice
Source: components/esphome/__init__.py:514
Integration: Automation (documentation, issues)
First occurred: March 27, 2023 at 12:25:53 AM (320 occurrences)
Last logged: 12:47:33 PM

espnspoffice automation: Choose at step 2: choice 16: Error executing script. Unexpected error for call_service at pos 2: Connection not done for espnspoffice @ 172.22.66.122; current state is ConnectionState.INITIALIZED!
espnspoffice automation: Choose at step 2: choice 30: Error executing script. Unexpected error for call_service at pos 2: Connection not done for espnspoffice @ 172.22.66.122; current state is ConnectionState.INITIALIZED!
espnspoffice automation: Choose at step 2: choice 18: Error executing script. Unexpected error for call_service at pos 2: Connection not done for espnspoffice @ 172.22.66.122; current state is ConnectionState.INITIALIZED!
espnspoffice automation: Error executing script. Unexpected error for choose at pos 2: Connection not done for espnspoffice @ 172.22.66.122; current state is ConnectionState.INITIALIZED!
While executing automation automation.espnspoffice
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 451, in _async_step
    await getattr(self, handler)()
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 684, in _async_call_service_step
    await service_task
  File "/usr/src/homeassistant/homeassistant/core.py", line 1808, in async_call
    task.result()
  File "/usr/src/homeassistant/homeassistant/core.py", line 1845, in _execute_service
    await cast(Callable[[ServiceCall], Awaitable[None]], handler.job.target)(
  File "/usr/src/homeassistant/homeassistant/components/esphome/__init__.py", line 514, in execute_service
    await entry_data.client.execute_service(service, call.data)
  File "/usr/local/lib/python3.10/site-packages/aioesphomeapi/client.py", line 1156, in execute_service
    self._check_authenticated()
  File "/usr/local/lib/python3.10/site-packages/aioesphomeapi/client.py", line 295, in _check_authenticated
    self._check_connected()
  File "/usr/local/lib/python3.10/site-packages/aioesphomeapi/client.py", line 289, in _check_connected
    raise APIConnectionError(
aioesphomeapi.core.APIConnectionError: Connection not done for espnspoffice @ 172.22.66.122; current state is ConnectionState.INITIALIZED!

**Comment by deejaybeam** [comment]
> @deejaybeam, At some point all of my devices started showing offline in esphome even though they were online and I could control them, view logs, and wirelessly install updates. Not sure what caused it. It showed up after an update to esphome. After some Googling I found some other people reporting similar issues and they fixed it either by using static addresses or with use_address.

This is just an issue with mdns, it´s normal e.g. if you use different networs for HA and esphom-devices.
In this case you just have to set: "status_use_ping: true" in your esphome-addon-configuration:
![image](https://user-images.githubusercontent.com/10434729/228509685-ac790023-c30d-4515-8fac-dc7a34dc15ae.png)

Thats exactly what this option is designed for.

Regarding to your error-messages:
While a reboot/restart - and also of HA itself it´s more/less normal that there are network interuptions.
Also HA side has the last state connected (..current state is ConnectionState.INITIALIZED!) but if there is a short network-interuption while HA calls a Service to write to the Panel - you see of course an error like this "APIConnectionError: Connection not done for espnspoffice @ 172.22.66.122; current state is ConnectionState.INITIALIZED!"

BUT this should not occour while "running", but I´m afraid this could be caused from your configurtaion with use-address!
Esphome wants to connect to the name via mdns, but this does not work, so it switches to use-address - but one of the next calls again tries to use the device-name, .....

PLEASE - try to set a static ip in esphome-configuration and remove use-address - this is also recommended in esphom-documentation - not in any forum found via google:
https://www.esphome.io/guides/faq.html#my-node-keeps-reconnecting-randomly
2nd point: Give your ESP a [static IP](https://www.esphome.io/components/wifi.html#wifi-manual-ip).

**Comment by rogerjak** [comment]
Upgraded from 3.2.1 to 3.2.2 and also stuck on "loading",
Just did the first step, updating tft, as the instructions said :(

**Comment by scutler81** [comment]
Sorry for the slow response. I've been away for work. I dropped the use_address and switched over to use static address as suggested and was still stuck.

Just in case I also updated the yaml and tft to 3.2.2. I also noticed that the blueprint had also changed so I deleted the old blueprint imported the latest version (and automations based on it). Set up all the entries and was delighted when my test panel suddenly came to life. 

I'll be slowly adding more and more buttons and sensors to the panel but if this works I'll be repeating the process on my other panels.

Thanks again for all the help!
